src/stores/RoomViewStore.js

```diff
@@ -128,16 +137,36 @@ class RoomViewStore extends Store {
         this._setState({
             joining: true,
         });
-        MatrixClientPeg.get().joinRoom(this._state.roomId, payload.opts).then(
-        () => {
-            this._setState({
-                joining: false,
+        MatrixClientPeg.get().joinRoom(this._state.roomId, payload.opts).done(() => {
+            dis.dispatch({
+                action: 'joined_room',
+                room_id: this._state.roomId,
             });
         }, (err) => {
-            this._setState({
-                joining: false,
-                joinError: err,
+            dis.dispatch({
+                action: 'join_room_error',
+                room_id: this._state.roomId,
```

Comment:
Agreed that nothing reads this error, but if it's there then it should probably be kept up to date. We could remove the join error though (unless we need it in RoomView by moving the Modal there)